src/winsvc.c

```diff
@@ -172,28 +172,116 @@ static int lua_EndService(lua_State *L) {
   return 0;
 }
 
+static int table_to_ServiceStatus(lua_State *L, SERVICE_STATUS *status) {
+  memset(status, 0, sizeof(SERVICE_STATUS));
+  if (lua_isnil(L, -1)) {
+    return 0;
+  }
+  if (!lua_istable(L, -1)) {
```

Comment:
instead of the lua_isnil on 177 and the istable on 180, try luaL_checktable(L, -1);... it does the same thing